src/core/opamConsole.ml

```diff
@@ -496,19 +496,28 @@ let timestamp () =
     (int_of_float (1000.0 *. msec))
 
 let log section ?(level=1) fmt =
-  if level <= OpamCoreConfig.(!r.debug_level) then
+  let debug_level =
+    let debug_level = OpamCoreConfig.(!r.debug_level) in
+    let sections = OpamCoreConfig.(!r.debug_sections) in
+    if OpamCoreConfig.StringSet.is_empty sections || OpamCoreConfig.StringSet.mem section sections then
+      debug_level
+    else
+      0
+  in
+  if level <= abs debug_level then
     let () = clear_status () in
+    let timestamp = if debug_level < 0 then "" else timestamp () in
```

Comment:
```suggestion
    let timestamp = if debug_level < 0 then "" else (timestamp ()) ^ "  " in 
```

In order to not have remaining spaces at the beginning of the line. But this implies to update the tests/patcher-test.reference file.

Reply:
Done!